crates/sui-adapter/src/temporary_store.rs

```diff
@@ -349,12 +366,56 @@ impl<S> Storage for TemporaryStore<S> {
 
         // For object deletion, we increment their version so that they will
         // eventually show up in the parent_sync table with an updated version.
-        self.deleted.insert(*id, (version.increment(), kind));
+        self.deleted
+            .insert(*id, (version.increment(), child_count, kind));
     }
 
     fn log_event(&mut self, event: Event) {
         self.events.push(event)
     }
+
+    fn change_child_counts(&mut self, deltas: BTreeMap<ObjectID, i64>) -> Result<(), ObjectID> {
+        let deltas = deltas
+            .into_iter()
+            .filter(|(_, delta)| *delta != 0)
+            .collect::<BTreeMap<_, _>>();
+        for id in deltas.keys() {
+            if !(self.written.contains_key(id) || self.deleted.contains_key(id)) {
+                let mut object = self.objects[id].clone();
+                // Active input object must be Move object.
+                object.data.try_as_move_mut().unwrap().increment_version();
+                self.written.insert(*id, object);
+            }
+        }
+        for (id, delta) in deltas {
+            match (self.written.get_mut(&id), self.deleted.get_mut(&id)) {
```

Comment:
Having trouble wrapping my head around this match, some comments would help

Reply:
Added comments